### ISSUE TITLE
Add watch support for possibly empty directories

### ISSUE THREAD

**Issue by MoonSweep** [saltstack/salt#24436]
Requisites can use glob for matching (#663, #678), but if a directory is empty, the requisite ends up with an error.

Example:

Debian provides a "sysfsutils" package with a startup script to set sysfs variables. They are stored in both /etc/sysfs.conf and /etc/sysfs.d/*. So I wrote this in an SLS:

``` yaml
sysfsutils:
  service.running:
    - name: sysfsutils
    - watch:
      - file: /etc/sysfs.conf
      {% if grains['osrelease'] >= '8' %}
      - file: /etc/sysfs.d/*
      {% endif %}
```

_Note: support for configuration snippets in /etc/sysfs.d was added in Jessie, hence the check on `grains['osrelease']`._

But on some hosts, the /etc/sysfs.d/ directory is empty, and I have this error:

``` yaml
          ID: sysfsutils
    Function: service.running
      Result: False
     Comment: The following requisites were not found:
                                 watch:
                                     file: /etc/sysfs.d/*
     Started: 
    Duration: 
     Changes:   
```

Please add a way to watch/require files in a directory, even if said directory is empty.

Thanks !

**Comment by jfindlay** [comment]
@MoonSweep, thanks for the feature request.  The directory being there (even if empty) should qualify for a watch.

**Comment by anlutro** [comment]
The issue here is that the watch glob matches on state IDs, not directory/file names, so if there are no states matching `/etc/sysfs.d/*` it errors. I think it should just silently do nothing if no states with the glob is found.

**Comment by suslikas** [comment]
Also look to new bug with i fond, https://github.com/saltstack/salt/issues/663 not work if directory have two dots, like 

```
file: /etc/zabbix/zabbix_agentd.conf.d/*.conf
```

**Comment by cachedout** [comment]
@dmurphy18 Why was the Boron label applied to this issue?

**Comment by onlyanegg** [comment]
Is there any update on this. Still seeing @MoonSweep's issue on 2016.3.2 (Boron).

          ID: start_logstash
    Function: service.running
      Result: False
     Comment: The following requisites were not found:
                                 watch:
                                     file: /etc/logstash/conf.d/*
     Started:
    Duration:
     Changes:

Do I need to have a state managing these files?

**Comment by OrangeDog** [comment]
Bump to reopen this.

**Comment by OrangeDog** [comment]
> the watch glob matches on state IDs, not directory/file names

So I should instead be doing something like
```yaml
onchanges:
  '/usr/local/share/ca-certificates*'
```

For some reason the globbing isn't documented
https://docs.saltstack.com/en/latest/ref/states/requisites.html#requisite-matching

**Comment by OrangeDog** [comment]
This is still either a feature request and/or a documentation defect.

**Comment by whytewolf** [comment]
closing this as the original question in the ticket is not how requisites work. and this is more of a documentation issue that needs its own ticket.

**Comment by OrangeDog** [comment]
That documentation issue was #52308, and fixed in #62928.
